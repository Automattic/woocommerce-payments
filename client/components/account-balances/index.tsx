/**
 * External dependencies
 */
import React from 'react';
import { Card } from '@wordpress/components';

/**
 * Internal dependencies
 */
import AccountBalancesHeader from './header';
import './style.scss';
import AccountBalancesTabPanel from './balances-tab-panel';

/**
 * Renders an overview of the account's balances.
 *
 * @return {JSX.Element} Rendered element with the account balances card.
 */
const AccountBalances: React.FC = () => {
	return (
		<Card className="wcpay-account-balances">
			<AccountBalancesHeader />

			<AccountBalancesTabPanel />
		</Card>
	);
};

export default AccountBalances;
