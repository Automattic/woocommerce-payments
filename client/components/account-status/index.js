/** @format **/

/**
 * External dependencies
 */
import {
	Button,
	Card,
	CardBody,
	CardHeader,
	FlexBlock,
	FlexItem,
} from '@wordpress/components';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import AccountFees from './account-fees';
import AccountStatusItem from './account-status-item';
import DepositsStatus from 'components/deposits-status';
import PaymentsStatus from 'components/payments-status';
import StatusChip from './status-chip';
import './style.scss';
import './shared.scss';

const AccountStatusCard = ( props ) => {
	const { title, children, value } = props;
	return (
		<Card isMedium>
			<CardHeader
				className={ 'woocommerce-account-status__header' }
				direction={ 'row' }
				align={ 'center' }
				justify={ 'left' }
				gap={ 3 }
				expanded
			>
				{ title }
			</CardHeader>
			<CardBody>{ children || value || null }</CardBody>
		</Card>
	);
};

const AccountStatusError = () => {
	const cardTitle = __( 'Account details', 'woocommerce-payments' );
	return (
		<AccountStatusCard title={ cardTitle }>
			{ __(
				'Error determining the connection status.',
				'woocommerce-payments'
			) }
		</AccountStatusCard>
	);
};

const AccountStatusDetails = ( props ) => {
	const { accountStatus, accountFees } = props;

	const cardTitle = (
		<>
			<FlexItem className={ 'account-details' }>
				{ __( 'Account details', 'woocommerce-payments' ) }
			</FlexItem>
			<FlexBlock className={ 'account-status' }>
				<StatusChip accountStatus={ accountStatus.status } />
			</FlexBlock>
			<FlexItem className={ 'edit-details' }>
				<Button isLink href={ accountStatus.accountLink }>
					{ __( 'Edit details', 'woocommerce-payments' ) }
				</Button>
			</FlexItem>
		</>
	);

	return (
		<AccountStatusCard title={ cardTitle }>
			<AccountStatusItem
				label={ __( 'Payments:', 'woocommerce-payments' ) }
			>
				<PaymentsStatus
					paymentsEnabled={ accountStatus.paymentsEnabled }
				/>
			</AccountStatusItem>
			<AccountStatusItem
				label={ __( 'Deposits:', 'woocommerce-payments' ) }
			>
				<DepositsStatus
					status={ accountStatus.deposits?.status }
					interval={ accountStatus.deposits?.interval }
				/>
			</AccountStatusItem>
			{ 0 < accountFees.length && (
				<AccountFees accountFees={ accountFees } />
			) }
		</AccountStatusCard>
	);
};

const AccountStatus = ( props ) => {
	const { accountStatus } = props;
	return accountStatus.error ? (
		<AccountStatusError />
	) : (
		<AccountStatusDetails { ...props } />
	);
};

export default AccountStatus;
