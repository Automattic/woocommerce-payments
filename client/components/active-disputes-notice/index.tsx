/**
 * External dependencies
 */
import React from 'react';
import { __ } from '@wordpress/i18n';
import { TaskItem } from '@woocommerce/experimental';

/**
 * Internal dependencies
 */

import { useDisputes } from 'wcpay/data';
import { getDisputesNoticeString } from 'wcpay/disputes/utils';
import { getAdminUrl } from 'wcpay/utils';
import './style.scss';

const ActiveDisputesNotice: React.FC = () => {
	const { disputes: activeDisputes, isLoading } = useDisputes( {
		filter: 'awaiting_response',
	} );

	if ( ! activeDisputes || isLoading ) {
		return null;
	}

	const disputesNoticeString = getDisputesNoticeString( activeDisputes );

	if ( ! disputesNoticeString ) {
		return null;
	}

	const handleClick = () => {
		window.location.href = getAdminUrl( {
			page: 'wc-admin',
			path: '/payments/disputes',
			filter: 'awaiting_response',
		} );
	};

	return (
		<div className="wcpay-active-disputes-notice">
			<TaskItem
				title={ disputesNoticeString }
				expanded
				completed={ false }
				content={ '' } // TODO: add subtitle here
				level={ 2 } // TODO: dynamic level dispute deadline is approaching
				showActionButton
				actionLabel={ __( 'Respond now', 'woocommerce-payments' ) }
				onClick={ handleClick }
				action={ handleClick }
			/>
		</div>
	);
};

export default ActiveDisputesNotice;
