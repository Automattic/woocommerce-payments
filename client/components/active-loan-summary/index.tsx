/**
 * External dependencies
 */
import * as React from 'react';
import {
	Button,
	Card,
	CardBody,
	CardHeader,
	Flex,
	FlexBlock,
	FlexItem,
} from '@wordpress/components';
import { __, sprintf } from '@wordpress/i18n';
import { createInterpolateElement } from '@wordpress/element';
import { dateI18n } from '@wordpress/date';

/**
 * Internal dependencies.
 */
import { formatExplicitCurrency } from 'utils/currency';
import Loadable from 'components/loadable';
import { useActiveLoanSummary } from 'wcpay/data';

import './style.scss';

const Block = ( {
	title,
	children,
}: {
	title: React.ReactNode;
	children: React.ReactNode;
} ): JSX.Element => (
	<FlexBlock className="wcpay-loan-summary-block">
		<div className="wcpay-loan-summary-block__title">{ title }</div>
		<div className="wcpay-loan-summary-block__value">{ children }</div>
	</FlexBlock>
);

const ActiveLoanSummaryLoading = (): JSX.Element => {
	return (
		<Card>
			<CardHeader size="medium" className="wcpay-loan-summary-header">
				<FlexItem>
					{ __( 'Active loan overview', 'woocommerce-payments' ) }
				</FlexItem>
			</CardHeader>
			<CardBody className="wcpay-loan-summary-body">
				<Flex align="normal" className="wcpay-loan-summary-row">
					<Block
						title={ __( 'Total repaid', 'woocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Total repaid placeholder"
						/>
					</Block>
					<Block
						title={ __(
							'Repaid this period',
							'woocommerce-payments'
						) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Repaid this period placeholder"
						/>
					</Block>
				</Flex>
				<Flex
					align="normal"
					className="wcpay-loan-summary-row is-bottom-row"
				>
					<Block
						title={ __( 'Loan disbursed', 'woocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Date disbursed"
						/>
					</Block>
					<Block
						title={ __( 'Loan amount', 'woocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Loan amount"
						/>
					</Block>
					<Block title={ __( 'Fixed fee', 'woocommerce-payments' ) }>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Fixed fee"
						/>
					</Block>
					<Block
						title={ __( 'Withhold rate', 'woocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="Rate"
						/>
					</Block>
					<Block
						title={ __( 'First paydown', 'woocommerce-payments' ) }
					>
						<Loadable
							isLoading={ true }
							display="inline"
							placeholder="First paydown"
						/>
					</Block>
				</Flex>
			</CardBody>
		</Card>
	);
};

const ActiveLoanSummary = (): JSX.Element => {
	const { summary, isLoading } = useActiveLoanSummary();

	if ( isLoading || ! summary ) {
		return <ActiveLoanSummaryLoading />;
	}

	const { details } = summary;

	return (
		<Card>
			<CardHeader size="medium" className="wcpay-loan-summary-header">
				<FlexItem>
					{ __( 'Active loan overview', 'woocommerce-payments' ) }
				</FlexItem>
				<FlexItem>
					<Button
						isLink
						href="javascript:alert('Not implemented yet')"
					>
						{ __( 'View transactions', 'woocommerce-payments' ) }
					</Button>
				</FlexItem>
			</CardHeader>
			<CardBody className="wcpay-loan-summary-body">
				<Flex align="normal" className="wcpay-loan-summary-row">
					<Block
						title={ __( 'Total repaid', 'woocommerce-payments' ) }
					>
						{ createInterpolateElement(
							sprintf(
								__(
									'<big>%s</big> of %s',
									'woocommerce-payments'
								),
								formatExplicitCurrency(
									details.paid_amount,
									details.currency
								),
								formatExplicitCurrency(
									details.fee_amount + details.advance_amount,
									details.currency
								)
							),
							{
								big: <span className="is-big" />,
							}
						) }
					</Block>
					<Block
						title={ sprintf(
							__(
								'Repaid this period (until %s)',
								'woocommerce-payments'
							),
							dateI18n(
								'M j, Y',
								new Date(
									details.current_repayment_interval.due_at *
										1000
								)
							)
						) }
					>
						{ createInterpolateElement(
							sprintf(
								__(
									'<big>%s</big> of %s minimum',
									'woocommerce-payments'
								),
								formatExplicitCurrency(
									details.current_repayment_interval
										.paid_amount,
									details.currency
								),
								formatExplicitCurrency(
									details.current_repayment_interval
										.paid_amount +
										details.current_repayment_interval
											.remaining_amount,
									details.currency
								)
							),
							{
								big: <span className="is-big" />,
							}
						) }
					</Block>
				</Flex>
				<Flex
					align="normal"
					className="wcpay-loan-summary-row is-bottom-row"
				>
					<Block
						title={ __( 'Loan disbursed', 'woocommerce-payments' ) }
					>
						{ dateI18n(
							'M j, Y',
							new Date( details.advance_paid_out_at * 1000 )
						) }
					</Block>
					<Block
						title={ __( 'Loan amount', 'woocommerce-payments' ) }
					>
						{ formatExplicitCurrency(
							details.advance_amount,
							details.currency
						) }
					</Block>
					<Block title={ __( 'Fixed fee', 'woocommerce-payments' ) }>
						{ formatExplicitCurrency(
							details.fee_amount,
							details.currency
						) }
					</Block>
					<Block
						title={ __( 'Withhold rate', 'woocommerce-payments' ) }
					>
						{ details.withhold_rate * 100 }%
					</Block>
					<Block
						title={ __( 'First paydown', 'woocommerce-payments' ) }
					>
						{ dateI18n(
							'M j, Y',
							new Date( details.repayments_begin_at * 1000 )
						) }
					</Block>
				</Flex>
			</CardBody>
		</Card>
	);
};

export default ActiveLoanSummary;
