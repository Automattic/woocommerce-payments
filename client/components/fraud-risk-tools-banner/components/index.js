/**
 * Internal dependencies
 */
import NewPill from './new-pill';
import BannerBody from './banner-body';
import BannerActions from './banner-actions';

export { NewPill, BannerBody, BannerActions };
