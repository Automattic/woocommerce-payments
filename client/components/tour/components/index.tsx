/**
 * Internal dependencies
 */
import TourStep from './TourStep';
import TourImage from './TourImage';
import TourFooter from './TourFooter';
import TourContent from './TourContent';
import TourNextButton from './TourNextButton';
import TourPreviousButton from './TourPreviousButton';

export {
	TourStep,
	TourImage,
	TourFooter,
	TourContent,
	TourNextButton,
	TourPreviousButton,
};
