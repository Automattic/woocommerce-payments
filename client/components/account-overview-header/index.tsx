/**
 * External dependencies
 */
import React from 'react';
import { CardHeader, Flex, FlexItem } from '@wordpress/components';

/**
 * Internal dependencies
 */
import { getGreeting } from './utils';
import { useCurrentWpUser } from './hooks';
import wooPaymentsLogo from 'assets/images/woopayments.svg?asset';
import './style.scss';

/**
 * Renders the header of the account's balances card.
 *
 * @return {JSX.Element} Rendered element with the account balances card header.
 */
const AccountOverviewHeader: React.FC = () => {
	const { user } = useCurrentWpUser();
	const greeting = getGreeting( user?.first_name );

	return (
		<CardHeader className="wcpay-account-overview-header">
			<Flex
				align="center"
				justify="space-between"
				className="wcpay-account-overview-header__flex"
			>
				<FlexItem className="wcpay-account-overview-header__flex__greeting">
					{ greeting }
				</FlexItem>
				<FlexItem>
					<img
						className="wcpay-account-overview-header__flex__logo"
						src={ wooPaymentsLogo }
						alt="Woo Payments logo"
						width={ 107 }
					/>
				</FlexItem>
			</Flex>
		</CardHeader>
	);
};

export default AccountOverviewHeader;
