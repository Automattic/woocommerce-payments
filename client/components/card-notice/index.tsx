/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import './styles.scss';
import { CardFooter } from '@wordpress/components';

interface CardNoticeProps {
	actions: JSX.Element;
}

const CardNotice: React.FC< CardNoticeProps > = ( { children, actions } ) => {
	return (
		<CardFooter className="card-notice">
			<div className="card-notice__section">
				<div className="card-notice__text">{ children }</div>
				<div className="card-notice__button">{ actions }</div>
			</div>
		</CardFooter>
	);
};

export default CardNotice;
