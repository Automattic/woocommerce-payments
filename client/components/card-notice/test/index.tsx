/** @format */
/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import React from 'react';

/**
 * Internal dependencies
 */
import CardNotice from '../';
import { Button } from '@wordpress/components';

describe( 'CardNotice component', () => {
	test( 'should render child', () => {
		const noticeText = 'Notice text';
		const { container, getByText } = render(
			<CardNotice>{ noticeText }</CardNotice>
		);

		expect( getByText( noticeText ) ).toBeInTheDocument();
		expect( container ).toMatchSnapshot();
	} );

	test( 'should render a button when actions prop is provided', () => {
		const buttonText = 'Notice text';

		const actions = <Button>{ buttonText }</Button>;
		const { container, getByRole } = render(
			<CardNotice actions={ actions }>{ 'Notice text' }</CardNotice>
		);

		expect(
			getByRole( 'button', {
				name: buttonText,
			} )
		).toBeInTheDocument();
		expect( container ).toMatchSnapshot();
	} );
} );
