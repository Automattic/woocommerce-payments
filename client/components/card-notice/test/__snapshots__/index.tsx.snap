// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardNotice component should render a button when actions prop is provided 1`] = `
<div>
  <div
    class="components-flex components-card__footer components-card-footer card-notice css-p1v47q-View-Flex-sx-Base-sx-Items-ItemsRow-Footer-borderRadius-borderColor-medium em57xhy0"
    data-wp-c16t="true"
    data-wp-component="CardFooter"
  >
    <div
      class="card-notice__section"
    >
      <div
        class="card-notice__text"
      >
        Notice text
      </div>
      <div
        class="card-notice__button"
      >
        <button
          class="components-button"
          type="button"
        >
          Notice text
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`CardNotice component should render child 1`] = `
<div>
  <div
    class="components-flex components-card__footer components-card-footer card-notice css-p1v47q-View-Flex-sx-Base-sx-Items-ItemsRow-Footer-borderRadius-borderColor-medium em57xhy0"
    data-wp-c16t="true"
    data-wp-component="CardFooter"
  >
    <div
      class="card-notice__section"
    >
      <div
        class="card-notice__text"
      >
        Notice text
      </div>
      <div
        class="card-notice__button"
      />
    </div>
  </div>
</div>
`;
